riot_pal/dut_pyshell.py

```diff
@@ -0,0 +1,258 @@
+
+        """
+        try:
+            results = self.dut.send_cmd(arg)
+        except KeyError as exc:
+            print('Could not parse argument {}'.format(exc))
+        except (TypeError, ValueError, SyntaxError) as exc:
+            print(exc)
+        else:
+            self._print_func_result_success(results)
+
+    def complete_send_cmd(self, text, line, begidx, endidx):
+        """Completes arg with memory map record names"""
+        begidx = begidx
+        endidx = endidx
+        return self._complete_cmd_list(text, line)
+
+    def _complete_cmd_list(self, text, line):
+        mline = line.partition(' ')[2]
+        offs = len(mline) - len(text)
+        map_records = self.cmd_list
+        return [s[offs:] for s in map_records if s.startswith(mline)]
+
+    def do_show_pinout(self, arg):
+        """Prints the pinout for the connected board
+
+        Usage:
+            show_pinout
+        """
+        print("""
+____________$$$$$$$$$$$$
+__________$$___________$$$
+_________$$_______________$$____________$$$$
+_______$$___________________$$______$$$$___$
+______$$___________________$$$$$___$$$____$
+_____$$__________________$$$___$$$$______$
+____$$________________$$$_____$__$_______$
+___$$_______________$$$____$$___$$$$____$
+___$$______________$$____$_____$$__$$___$
+___$$____________$$___$$_______$_$$$_$_$
+___$$___________$$_$$$_$$$$$____$$$$__$$
+___$$__________$_$$$$_$$__$$$$__$$$____$
+____$$________$$§§§§__$__$$$$$_____$$___$
+_____$$______$$§$§§$_____$$$_______$$$___$
+______$$___$$$§§$§§$$_____________________$
+_______$$_$$$§§$§§$§§$______________$$$$__ $
+_________$$$§§$$§$$§§$_____________$__$$__ $
+_______§$§$§§$$§§$§§§$____________$$$$$___ $
+______§$§$$§$§§§$$§§$_____________________ $
+_____$$§$$$$§§§$$$§$$$$___________________$
+____$$§$$$$§§§$$§$§§$$$$$$______________$$
+____$§§$§$§§§§$§§$§§$$$$___$$$$$$$$$$$$$
+___$$§$§§§§§§$§§§$§§$$$______________$
+___$§$$§§§§§$§§§§$§§$$$$____________$
+____$$§§§§§$§§§§§$§§$$$$____________$
+_____$§§§$$$§§§§§$§§$$$$$$$________$
+____$$§§$§§$§§§§$§§$$$$$$_$________$
+____$§§§§§§$§§§§$§§$$$_____$$_____$
+___$§§§§§§$§§§§$$§$$__________$$_$$
+$$$$§§§§§$$§§§§$§§$_____________$$
+$$§§§§§§$$$$§§$§§$$_______________$
+$§§§§$$§§§§§§$§§$__________________$$
+$§$$§§§§$§§$$§$$____________________$
+$$§$§§§$$§§$§$$$____________________$$
+$$$$§§§§§§§$$$$_____________________$$$
+§$$§§§$§§§§$$$$_____________________$$$
+__$$§§§§$§$§$_______________________$$$
+
+""")
+
+    def do_exit(self, arg):
+        """I mean it should be obvious
+
+        Usage:
+            exit
+        """
+        arg = arg
+        return True
+
+    def _print_func_result_success(self, results):
+        if not isinstance(results, list):
+            results = [results]
+        result = RESULT_SUCCESS
+        printed_something = False
+        for res in results:
+            if self.data_only:
+                if 'result' in res:
+                    if res['result'] == RESULT_SUCCESS:
```

Comment:
why this distinction, if there was an error data should be empty and printed accordingly.

Reply:
I am not super sure about that.  How does one get error codes?  It is all about filtering only the minimum and most useful data.  One can turn off the filter if they need to know the error code.